timelock-impl/src/test/java/com/palantir/atlasdb/timelock/paxos/BatchingPaxosLatestSequenceCacheTests.java

```diff
@@ -0,0 +1,173 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.paxos;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.entry;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+import java.util.Map;
+import java.util.Optional;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.palantir.common.streams.KeyedStream;
+import com.palantir.conjure.java.api.errors.ErrorType;
+import com.palantir.conjure.java.api.errors.RemoteException;
+import com.palantir.conjure.java.api.errors.SerializableError;
+import com.palantir.conjure.java.api.errors.ServiceException;
+import com.palantir.paxos.ImmutablePaxosLong;
+import com.palantir.paxos.PaxosLong;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BatchingPaxosLatestSequenceCacheTests {
+
+    private static final Client CLIENT_1 = Client.of("client-1");
+    private static final Client CLIENT_2 = Client.of("client-2");
+    private static final Client CLIENT_3 = Client.of("client-3");
+    private static final Map<Client, PaxosLong> INITIAL_UPDATES = ImmutableMap.<Client, PaxosLong>builder()
+            .put(CLIENT_1, PaxosLong.of(5L))
+            .put(CLIENT_2, PaxosLong.of(10L))
+            .put(CLIENT_3, PaxosLong.of(15L))
+            .build();
+
+    @Mock
+    private BatchPaxosAcceptor remote;
+
+    @Test
+    public void withoutCacheKeyWeGetEverything() {
+        AcceptorCacheDigest digest = ImmutableAcceptorCacheDigest.builder()
+                .newCacheKey(AcceptorCacheKey.newCacheKey())
+                .putUpdates(CLIENT_1, 5)
+                .putUpdates(CLIENT_2, 10)
+                .build();
+
+        when(remote.latestSequencesPreparedOrAccepted(Optional.empty(), ImmutableSet.of(CLIENT_1, CLIENT_2)))
+                .thenReturn(digest);
+
+        BatchingPaxosLatestSequenceCache cache = new BatchingPaxosLatestSequenceCache(remote);
+        Map<Client, PaxosLong> result = cache.apply(ImmutableSet.of(CLIENT_1, CLIENT_2));
+
+        assertThat(result).containsOnly(
+                entry(CLIENT_1, PaxosLong.of(5L)),
+                entry(CLIENT_2, PaxosLong.of(10L)));
+    }
+
+    @Test
+    public void returnsSameResultIfCached() {
+        BatchingPaxosLatestSequenceCache cache = initialCache();
```

Comment:
`@Before` pl0x

Reply:
meh, not a huge fan, I'm able to see everything just by looking at the test, and all this saves me is one line 🤷‍♂